File: Java OOP/Homework07/presenters/ProgrammerPresenter.py
from models import Programmer
from views import View


class ProgrammerPresenter:
    def __init__(self, programmer: Programmer, view: View):
        self.programmer = programmer
        self.view = view

    def print_programmer_info(self):
        self.view.print_message("-------Личная информация-------")
        self.view.print_message(f"Фамилия: {self.programmer.surname}")
        self.view.print_message(f"Имя: {self.programmer.name}")
        self.view.print_message(f"Отчество: {self.programmer.patronymic}")
        self.view.print_message(f"Телефон: {self.programmer.telephone}")
        self.view.print_message(f"Возраст: {self.programmer.age}")

        self.view.print_message("---Корпоративная информация---")
        self.view.print_message(f"Позиция: {self.programmer.qualification}")
        self.view.print_message(f"Зарплата: {self.programmer.salary}")

File: Java OOP/Homework06/presenters/ProgrammerPresenter.java
package presenters;

import models.Programmer;
import views.View;

public class ProgrammerPresenter {
    Programmer programmer;
    View view;

    public ProgrammerPresenter(Programmer programmer, View view) {
        this.programmer = programmer;
        this.view = view;
    }

    public void printProgrammerInfo() {        
        view.printMessage("-------Личная информация-------");
        view.printMessage(String.format("Фамилия: %s", programmer.getSurname()));
        view.printMessage(String.format("Имя: %s", programmer.getName()));
        view.printMessage(String.format("Отчество: %s",programmer.getPatronymic()));
        view.printMessage(String.format("Телефон: %s", programmer.getTelephone()));
        view.printMessage(String.format("Возраст: %d", programmer.getAge()));

        
        view.printMessage("---Корпоративная информация---");
        view.printMessage(String.format("Позиция: %s", programmer.getQualification()));
        view.printMessage(String.format("Зарплата: %d", programmer.getSalary()));
    }

}
